### PR TITLE
[node] Add 'end' property in PipelineOptions interface

### PR DESCRIPTION
**Pull request by xfournet**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/nodejs/node/pull/40886 and https://nodejs.org/docs/latest-v18.x/api/stream.html#streampipelinesource-transforms-destination-options
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/node/stream.d.ts [hunk 1]
```diff
@@ -1179,7 +1179,8 @@ declare module 'stream' {
             : (err: NodeJS.ErrnoException | null) => void;
         type PipelinePromise<S extends PipelineDestination<any, any>> = S extends PipelineDestinationPromiseFunction<any, infer P> ? Promise<P> : Promise<void>;
         interface PipelineOptions {
-            signal: AbortSignal;
+            signal?: AbortSignal;
+            end?: boolean;
```
Comment (Semigradsky) [inline types/node/stream.d.ts:1183]: Looks good!

Maybe also allowing to pass `undefined` in options for users with turned on [exactOptionalPropertyTypes](exactOptionalPropertyTypes) as in another options:
```ts
            signal?: AbortSignal | undefined;
            end?: boolean | undefined;
```

Comment (xfournet) [reply types/node/stream.d.ts:1183]: Yes you right, i updated it

**Review by Semigradsky — APPROVED** [review]

**Review by peterblazejewicz — APPROVED** [review]
@xfournet thanks!

**Comment by xfournet** [comment]
Ready to merge

**Pull request merged**